management/src/main/java/io/micronaut/management/health/indicator/jdbc/JdbcIndicator.java

```diff
@@ -60,54 +65,96 @@
     private static final int CONNECTION_TIMEOUT = 3;
 
     private final ExecutorService executorService;
-    private final DataSource[] dataSources;
-    private final DataSourceResolver dataSourceResolver;
+    private final Map<String, DataSource> dataSources;
     private final HealthAggregator<?> healthAggregator;
 
     /**
      * @param executorService    The executor service
      * @param dataSources        The data sources
      * @param dataSourceResolver The data source resolver
      * @param healthAggregator   The health aggregator
+     * @deprecated Use {@link #JdbcIndicator(java.util.concurrent.ExecutorService, java.util.Collection, io.micronaut.jdbc.DataSourceResolver, io.micronaut.management.health.aggregator.HealthAggregator)} instead
      */
+    @Deprecated
     public JdbcIndicator(@Named(TaskExecutors.IO) ExecutorService executorService,
                          DataSource[] dataSources,
                          @Nullable DataSourceResolver dataSourceResolver,
                          HealthAggregator<?> healthAggregator) {
         this.executorService = executorService;
-        this.dataSources = dataSources;
-        this.dataSourceResolver = dataSourceResolver != null ? dataSourceResolver : DataSourceResolver.DEFAULT;
+        Map<String, DataSource> dataSourceMap = new LinkedHashMap<>(dataSources.length);
+        for (int i = 0; i < dataSources.length; i++) {
+            DataSource dataSource = dataSources[i];
+            if (dataSourceResolver != null) {
+                dataSource = dataSourceResolver.resolve(dataSource);
+            }
+            dataSourceMap.put("dataSource " + (i + 1), dataSource);
+        }
+        this.dataSources = dataSourceMap;
+        this.healthAggregator = healthAggregator;
+    }
+
+    /**
+     * @param executorService    The executor service
+     * @param dataSources        The data sources
+     * @param dataSourceResolver The data source resolver
+     * @param healthAggregator   The health aggregator
+     */
+    @Inject
+    public JdbcIndicator(@Named(TaskExecutors.IO) ExecutorService executorService,
+                         Collection<BeanRegistration<DataSource>> dataSources,
+                         @Nullable DataSourceResolver dataSourceResolver,
+                         HealthAggregator<?> healthAggregator) {
+        this.executorService = executorService;
+        Map<String, DataSource> dataSourceMap = new LinkedHashMap<>(dataSources.size());
+        for (BeanRegistration<DataSource> dataSource : dataSources) {
+            DataSource ds = dataSource.bean();
+            if (dataSourceResolver != null) {
+                ds = dataSourceResolver.resolve(ds);
+            }
+            final BeanIdentifier id = dataSource.id();
+            if (id != null) {
+                dataSourceMap.put(id.getName(), ds);
+            } else {
+                dataSourceMap.put("default", ds);
```

Comment:
I think this could result in some datasources not being reported if the beans are created without a named qualifier

Reply:
A data source without a qualifier is the default datasource, if more than one datasource existed without a qualifier then you would get non unique bean exceptions all over the place prior to even hitting this code.

Reply:
Where would you get a non unique exception specifically? Is that with the assumption they are using our libraries for managing the creation of the beans?

Reply:
If you have more than one data source and no way to pick which one to use in your application then you must be a super hero regardless if you use our libraries or not

Reply:
Do we not allow injecting a single instance based on the highest order? Does that require super hero powers?